### PR TITLE
 Nuttx Duo S #80   Duo S buildroot to 1.1.3 #78

### PR DESCRIPTION
**Pull request by Gekyume777**
Update the latest message of Nuttx & DuoS， base on https://github.com/lupyuen/nuttx-sg2000
Updata the latest message of BuildRoot, version = Duo-V1.1.3,base on https://github.com/milkv-duo/duo-buildroot-sdk/releases

### CODE REVIEW

File: Duo_S/BuildRoot/README.md [hunk 1]
```diff
@@ -32,17 +32,20 @@ last_update: 2024-06-21
 
 Install the [`ruyi`](https://github.com/ruyisdk/ruyi) package manager, run `ruyi device provision`, and follow the prompts.
 
+### using `balenaEtcher` to Flash Image to microSD Card
```
Comment (KevinMX) [inline Duo_S/BuildRoot/README.md:35]:
```suggestion
### Using `balenaEtcher` to flash image to microSD Card
```

File: Duo_S/BuildRoot/README.md [hunk 2]
```diff
@@ -75,7 +78,7 @@ PRETTY_NAME="Buildroot 2021.05"
 
 Screen recording (From flashing image to login):
 
-[![asciicast](https://asciinema.org/a/Zbt8azPsJFYLWOYCKgPNrt9S7.svg)](https://asciinema.org/a/Zbt8azPsJFYLWOYCKgPNrt9S7)
+[!bilibili](https://www.bilibili.com/video/BV1kCU6Y1ECg/?vd_source=1291fcfaefa3ea5589519340cf2ed80c)(https://www.bilibili.com/video/BV1kCU6Y1ECg/?vd_source=1291fcfaefa3ea5589519340cf2ed80c)
```
Comment (KevinMX) [inline Duo_S/BuildRoot/README.md:81]: Use asciinema for CLI only records.
Bilibili is not really usable outside Chinese mainland.

File: Duo_S/BuildRoot/README.md [hunk 3]
```diff
@@ -32,17 +32,20 @@ last_update: 2024-06-21
 
 Install the [`ruyi`](https://github.com/ruyisdk/ruyi) package manager, run `ruyi device provision`, and follow the prompts.
 
+### using `balenaEtcher` to Flash Image to microSD Card
+Install[`balennaEtcher`],use open source tool balenaEtcher for flashing
```
Comment (KevinMX) [inline Duo_S/BuildRoot/README.md:36]: Not sure if this is what you intended to say...
```suggestion
Install `balennaEtcher`, the open source tool for flashing.
```

File: Duo_S/BuildRoot/README_zh.md [hunk 4]
```diff
@@ -20,21 +20,24 @@
 
 ## 安装步骤
 
-### 使用 `ruyi` CLI 刷写镜像到 microSD 卡
+### 1 使用 `ruyi` CLI 刷写镜像到 microSD 卡
 
 安装 [`ruyi`](https://github.com/ruyisdk/ruyi) 包管理器，运行 `ruyi device provision` 并按提示操作。
 
+### 2 使用 `balenaEtcher` 刷写镜像到 microSD 卡
+安装[`balennaEtcher`] 使用开源跨平台工具 balenaEtcher 进行刷写
```
Comment (KevinMX) [inline Duo_S/BuildRoot/README_zh.md:28]:
```suggestion
安装开源跨平台工具 `balenaEtcher` 进行刷写。
```

File: Duo_S/BuildRoot/README_zh.md [hunk 5]
```diff
@@ -20,21 +20,24 @@
 
 ## 安装步骤
 
-### 使用 `ruyi` CLI 刷写镜像到 microSD 卡
+### 1 使用 `ruyi` CLI 刷写镜像到 microSD 卡
```
Comment (KevinMX) [inline Duo_S/BuildRoot/README_zh.md:23]:
```suggestion
### 使用 `ruyi` CLI 刷写镜像到 microSD 卡
```

File: Duo_S/BuildRoot/README_zh.md [hunk 6]
```diff
@@ -20,21 +20,24 @@
...
 
 安装 [`ruyi`](https://github.com/ruyisdk/ruyi) 包管理器，运行 `ruyi device provision` 并按提示操作。
 
+### 2 使用 `balenaEtcher` 刷写镜像到 microSD 卡
```
Comment (KevinMX) [inline Duo_S/BuildRoot/README_zh.md:27]:
```suggestion
### 使用 `balenaEtcher` 刷写镜像到 microSD 卡
```

File: Duo_S/BuildRoot/README_zh.md [hunk 7]
```diff
@@ -67,7 +70,7 @@ PRETTY_NAME="Buildroot 2021.05"
 
 屏幕录像（从刷写镜像到登录系统）：
 
-[![asciicast](https://asciinema.org/a/Zbt8azPsJFYLWOYCKgPNrt9S7.svg)](https://asciinema.org/a/Zbt8azPsJFYLWOYCKgPNrt9S7)
+[!bilibili](https://www.bilibili.com/video/BV1kCU6Y1ECg/?vd_source=1291fcfaefa3ea5589519340cf2ed80c)(https://www.bilibili.com/video/BV1kCU6Y1ECg/?vd_source=1291fcfaefa3ea5589519340cf2ed80c)
```
Comment (KevinMX) [inline Duo_S/BuildRoot/README_zh.md:73]: Same as above.

File: Duo_S/NuttX/README.md [hunk 8]
```diff
@@ -22,13 +22,14 @@ last_update: 2024-06-21
...
-- A USB card reader
-- A USB to UART Debugger (e.g., CP2102, FT2232, etc. Be aware that WCH CH340/341 series will cause garbled text output, DO NOT USE)
-- Three DuPont wires
-- Ethernet access for TFTP Boot
+- A USB-A to C or USB C to C cable to power the development board
+- A microSD card for booting
+- A USB card reader for microSD burning
+- A USB to UART debugger (such as: CP2102, FT2232, etc., garbled characters will appear when using the CH340/341 series, which is an expected phenomenon)
+- Three Dupont lines for uart interface
+- Network cabling for optical fiber access (TFTP Boot cannot use USB network)
```
Comment (KevinMX) [inline Duo_S/NuttX/README.md:30]: What do you mean optical fiber?
There's no such port on the Duo S. Nor needed by NuttX.

File: Duo_S/NuttX/README.md [hunk 9]
```diff
@@ -22,13 +22,14 @@ last_update: 2024-06-21
...
-- A USB-A to C or USB C to C cable for powering the development board
-- A microSD card
-- A USB card reader
-- A USB to UART Debugger (e.g., CP2102, FT2232, etc. Be aware that WCH CH340/341 series will cause garbled text output, DO NOT USE)
-- Three DuPont wires
-- Ethernet access for TFTP Boot
+- A USB-A to C or USB C to C cable to power the development board
+- A microSD card for booting
+- A USB card reader for microSD burning
+- A USB to UART debugger (such as: CP2102, FT2232, etc., garbled characters will appear when using the CH340/341 series, which is an expected phenomenon)
```
Comment (KevinMX) [inline Duo_S/NuttX/README.md:28]: I'll suggest using the original translation.

File: Duo_S/NuttX/README.md [hunk 10]
```diff
@@ -22,13 +22,14 @@ last_update: 2024-06-21
...
+- Three Dupont lines for uart interface
```
Comment (KevinMX) [inline Duo_S/NuttX/README.md:29]:
```suggestion
- Three Dupont wires for UART
```

File: Duo_S/NuttX/README.md [hunk 11]
```diff
@@ -207,11 +212,7 @@ nsh>
 
 Screen recording:
 
-Part 1:
-[![asciicast](https://asciinema.org/a/8wvErVrySR04Ri18rPJ99qai9.svg)](https://asciinema.org/a/8wvErVrySR04Ri18rPJ99qai9)
-
-Part 2:
-[![asciicast](https://asciinema.org/a/loBvsK69TBtZmicjdzqHX2B9z.svg)](https://asciinema.org/a/loBvsK69TBtZmicjdzqHX2B9z)
+[!bilibili](https://www.bilibili.com/video/BV1RMUeYjELe/?spm_id_from=333.999.0.0)
```
Comment (KevinMX) [inline Duo_S/NuttX/README.md:215]: Same.

File: milkv-duos-sd-v1.1.3-2024-0930.img.zip
Comment (KevinMX) [inline milkv-duos-sd-v1.1.3-2024-0930.img.zip:1]: DO NOT upload binary files including system images to this repo as they will not be accepted.
This repo currently only provides test reports, metadata and the main matrix.

File: Duo_S/NuttX/README_zh.md [hunk 12]
```diff
@@ -195,11 +195,8 @@ nsh>
 
 屏幕录像：
 
-Part 1:
-[![asciicast](https://asciinema.org/a/8wvErVrySR04Ri18rPJ99qai9.svg)](https://asciinema.org/a/8wvErVrySR04Ri18rPJ99qai9)
+[!bilibili](https://www.bilibili.com/video/BV1RMUeYjELe/?spm_id_from=333.999.0.0)
```
Comment (KevinMX) [inline Duo_S/NuttX/README_zh.md:198]: Same.

File: Duo_S/NuttX/README_zh.md [hunk 13]
```diff
@@ -92,7 +90,9 @@ cd nuttx
 tools/configure.sh milkv_duos:nsh
 make -j$(nproc)
 ```
-
+构建过程中如果遇到文件丢失，可以通过menuconfig配置选项
+- 在(Top) → Library Routines → Select math library 选择是否使用数学库，及其路径
+- 在(Top) → Library Routines → Builtin libclang_rt.profile选择是否启用llvm
```
Comment (KevinMX) [inline Duo_S/NuttX/README_zh.md:95]: I suggest adding spaces between English and Chinese words to make them look tidy.
Not necessary though, it's up to you to decide.
Same applies to all reports.

File: Duo_S/NuttX/README_zh.md [hunk 14]
```diff
@@ -77,11 +75,11 @@ git clone https://github.com/apache/nuttx-apps apps
...
 Nuttx 需要使用 xPack 工具链进行编译，首先获取工具链，从 Github 下载，自行改变版本以匹配你的系统：
 ```bash
-wget https://github.com/xpack-dev-tools/riscv-none-elf-gcc-xpack/releases/download/v11.5.0-1/xpack-riscv-none-elf-gcc-13.3.0-1-linux-x64.tar.gz
-tar -xvzf xpack-riscv-none-elf-gcc-13.3.0-1-linux-x64.tar.gz
+wget https://github.com/xpack-dev-tools/riscv-none-elf-gcc-xpack/releases/download/v14.2.0-2/xpack-riscv-none-elf-gcc-14.2.0-2-linux-x64.tar.gz
+tar -xvzf xpack-riscv-none-elf-gcc-14.2.0-2-linux-x64.tar.gz
 ```
 
-将其添加到你的 PATH 中：
+/*建议将tar解压到/opt*/并将其添加到你的 PATH 中：
```
Comment (KevinMX) [inline Duo_S/NuttX/README_zh.md:82]: /* */
What's this?
We don't write quotes like this in Markdown.
Remove it if you don't want it.

**Review by KevinMX — CHANGES_REQUESTED** [review]
Too many issues.
This will not get merged unless all issues are fixed, and please s quash your commits.

File: Duo_S/NuttX/README.md (hunk 8 above)
Comment (Gekyume777) [reply Duo_S/NuttX/README.md:30]: Translation error.I will be caerful with translation software.

Comment (Gekyume777) [reply Duo_S/NuttX/README.md:30]: Translation error
I will be careful with translation software

File: Duo_S/NuttX/README_zh.md (hunk 14 above)
Comment (Gekyume777) [reply Duo_S/NuttX/README_zh.md:82]: Some stupid habits

**Pull request closed without merging**